### ISSUE TITLE
Error: Invalid parameter onlyif at /tmp/vagrant-puppet-3/manifests/nodes/Apache.pp:217

### ISSUE THREAD

**Issue by jcrosswh** [puphpet/puphpet#1539]
==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/composer/lib/facter/composer_home.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/concat/lib/facter/concat_basedir.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/epel/lib/facter/os_maj_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/firewall/lib/facter/ip6tables_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/firewall/lib/facter/iptables_persistent_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/firewall/lib/facter/iptables_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/git/lib/facter/git_exec_path.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/git/lib/facter/git_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/php/lib/facter/php_fact_extension_dir.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/php/lib/facter/php_fact_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/puppi/lib/facter/last_run.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/puppi/lib/facter/puppi_projects.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/puppi/lib/facter/windows_common_appdata.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/rabbitmq/lib/facter/rabbitmq_erlang_cookie.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/rvm/lib/facter/rvm_installed.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/rvm/lib/facter/rvm_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/staging/lib/facter/staging_http_get.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/staging/lib/facter/staging_windir.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/stdlib/lib/facter/facter_dot_d.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/stdlib/lib/facter/pe_version.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/stdlib/lib/facter/puppet_vardir.rb
    ==> default: Info: Loading facts in /tmp/vagrant-puppet-3/modules-0/stdlib/lib/facter/root_home.rb
    ==> default: Warning: Could not retrieve fact fqdn
    ==> default: Error: Invalid parameter onlyif at /tmp/vagrant-puppet-3/manifests/nodes/Apache.pp:217 on node packer-virtualbox-iso-1422601639
    ==> default: Error: Invalid parameter onlyif at /tmp/vagrant-puppet-3/manifests/nodes/Apache.pp:217 on node packer-virtualbox-iso-1422601639
    The SSH command responded with a non-zero exit status. Vagrant
    assumes that this means the command failed. The output for this command
    should be in the log above. Please read the output to determine what
    went wrong.

config.yaml:

    vagrantfile:
        target: local
        vm:
            box: puphpet/ubuntu1404-x64
            box_url: puphpet/ubuntu1404-x64
            hostname: ''
            memory: '512'
            cpus: '1'
            chosen_provider: virtualbox
            network:
                private_network: 192.168.56.101
                forwarded_port:
                    vflnp_6vzeaqdea2lz:
                        host: '7408'
                        guest: '22'
            post_up_message: ''
            provider:
                virtualbox:
                    modifyvm:
                        natdnshostresolver1: on
                vmware:
                    numvcpus: 1
                parallels:
                    cpus: 1
            provision:
                puppet:
                    manifests_path: puphpet/puppet
                    manifest_file: site.pp
                    module_path: puphpet/puppet/modules
                    options:
                        - '--verbose'
                        - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
                        - '--parser future'
            synced_folder:
                vflsf_7hdzzskqlww2:
                    source: ./
                    target: /var/www
                    sync_type: default
                    rsync:
                        args:
                            - '--verbose'
                            - '--archive'
                            - '-z'
                        exclude:
                            - .vagrant/
                            - .git/
                        auto: 'true'
                    owner: www-data
                    group: www-data
            usable_port_range:
                start: 10200
                stop: 10500
        ssh:
            host: null
            port: null
            private_key_path: null
            username: vagrant
            guest_port: null
            keep_alive: true
            forward_agent: false
            forward_x11: false
            shell: 'bash -l'
        vagrant:
            host: detect
    server:
        install: '1'
        packages: {  }
    users_groups:
        install: '1'
        groups: {  }
        users: {  }
    firewall:
        install: '1'
        rules: {  }
    cron:
        install: '1'
        jobs: {  }
    nginx:
        install: '0'
        settings:
            default_vhost: 1
            proxy_buffer_size: 128k
            proxy_buffers: '4 256k'
        upstreams: {  }
        vhosts:
            nxv_6cc3wfh0gb4u:
                server_name: awesome.dev
                server_aliases:
                    - www.awesome.dev
                www_root: /var/www/awesome
                listen_port: '80'
                index_files:
                    - index.html
                    - index.htm
                    - index.php
                client_max_body_size: 1m
                ssl: '0'
                ssl_cert: ''
                ssl_key: ''
                ssl_port: '443'
                rewrite_to_https: '1'
                spdy: '1'
                locations:
                    nxvl_551k5jynihsq:
                        location: /
                        autoindex: off
                        try_files:
                            - $uri
                            - $uri/
                            - /index.php$is_args$args
                        fastcgi: ''
                        fastcgi_index: ''
                        fastcgi_split_path: ''
                    nxvl_xodonya3ei8w:
                        location: '~ \.php$'
                        autoindex: off
                        try_files:
                            - $uri
                            - $uri/
                            - /index.php$is_args$args
                        fastcgi: '127.0.0.1:9000'
                        fastcgi_index: index.php
                        fastcgi_split_path: '^(.+\.php)(/.*)$'
                        fast_cgi_params_extra:
                            - 'SCRIPT_FILENAME $request_filename'
                            - 'APP_ENV dev'
        proxies: {  }
    apache:
        install: '1'
        settings:
            user: www-data
            group: www-data
            default_vhost: true
            manage_user: false
            manage_group: false
            sendfile: 0
        modules:
            - proxy_fcgi
            - rewrite
        vhosts: {  }
    php:
        install: '1'
        settings:
            version: '56'
        fpm_settings:
            port: 9000
        composer: '1'
        composer_home: ''
        modules:
            php:
                - cli
                - intl
                - mcrypt
                - sqlite
            pear: {  }
            pecl: {  }
        ini:
            display_errors: On
            error_reporting: '-1'
            session.save_path: /var/lib/php/session
            date.timezone: UTC
    xdebug:
        install: '0'
        settings:
            xdebug.default_enable: '1'
            xdebug.remote_autostart: '0'
            xdebug.remote_connect_back: '1'
            xdebug.remote_enable: '1'
            xdebug.remote_handler: dbgp
            xdebug.remote_port: '9000'
    xhprof:
        install: '0'
    wpcli:
        install: '0'
        version: 6.3.0
    drush:
        install: '0'
        version: null
    ruby:
        install: '1'
        versions: {  }
    python:
        install: '1'
        packages: {  }
        versions: {  }
    nodejs:
        install: '0'
        npm_packages: {  }
    hhvm:
        install: '0'
        nightly: 0
        composer: '1'
        composer_home: ''
        settings:
            host: 127.0.0.1
            port: '9000'
        ini:
            display_errors: On
            error_reporting: '-1'
        timezone: null
    mysql:
        install: '0'
        settings:
            version: '5.6'
            root_password: '123'
            override_options: {  }
        adminer: 0
        users:
            mysqlnu_m8b5eki2dl24:
                name: dbuser
                password: '123'
        databases:
            mysqlnd_r8fdhlh0prb3:
                name: dbname
                sql: ''
        grants:
            mysqlng_dasx6vzl69wo:
                user: dbuser
                table: '*.*'
                privileges:
                    - ALL
    postgresql:
        install: '0'
        settings:
            global:
                encoding: UTF8
                version: '9.3'
            server:
                postgres_password: '123'
        databases: {  }
        users: {  }
        grants: {  }
        adminer: 0
    mongodb:
        install: '0'
        settings:
            auth: 1
            bind_ip: 127.0.0.1
            port: '27017'
        databases: {  }
    redis:
        install: '0'
        settings:
            conf_port: '6379'
    sqlite:
        install: '1'
        adminer: 0
        databases: {  }
    mailcatcher:
        install: '0'
        settings:
            smtp_ip: 0.0.0.0
            smtp_port: 1025
            http_ip: 0.0.0.0
            http_port: '1080'
            mailcatcher_path: /usr/local/rvm/wrappers/default
            from_email_method: inline
    beanstalkd:
        install: '0'
        settings:
            listenaddress: 0.0.0.0
            listenport: '13000'
            maxjobsize: '65535'
            maxconnections: '1024'
            binlogdir: /var/lib/beanstalkd/binlog
            binlogfsync: null
            binlogsize: '10485760'
        beanstalk_console: 0
        binlogdir: /var/lib/beanstalkd/binlog
    rabbitmq:
        install: '0'
        settings:
            port: '5672'
        users: {  }
        vhosts: {  }
        plugins: {  }
    elastic_search:
        install: '0'
        settings:
            version: 1.4.1
            java_install: true
    solr:
        install: '0'
        settings:
            version: 4.10.2
            port: '8984'

**Comment by blazingarrow** [comment]
Same with nginx :flushed::
```
==> default: Error: Invalid parameter onlyif at /tmp/vagrant-puppet/manifests-893fcda21077f5a583c2dbbba8515c0a/nodes/Nginx.pp:309 on node cineboxx.ch
==> default: Error: Invalid parameter onlyif at /tmp/vagrant-puppet/manifests-893fcda21077f5a583c2dbbba8515c0a/nodes/Nginx.pp:309 on node cineboxx.ch
```

**Comment by jtreminio** [comment]
Reverted this commit 6 hours ago while I try to figure out another way around this.

**Comment by jtreminio** [comment]
Closed by 18d550e
